src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -1351,18 +1351,30 @@ protected function getIndexSqlDefinition(Index $index)
             $def .= ' `' . $index->getName() . '`';
         }
 
+        $columnNames = $index->getColumns();
+        $order = $index->getOrder() ?? [];
+        $columnNames = array_map(function ($columnName) use ($order) {
+            $ret = '`' . $columnName . '`';
+            if (isset($order[$columnName])) {
+                $ret .= ' ' . $order[$columnName];
+            }
+
+            return $ret;
+        }, $columnNames);
```

Comment:
Similar to how SRID is handled for the column definition for geometry types, we can gate keep whether or not to add the order clause through a `version_compare`.

```suggestion
        $orderSupported = version_compare($this->getAttribute(\PDO::ATTR_SERVER_VERSION), '8') > -1;
        $columnNames = array_map(function ($columnName) use ($order) {
            $ret = '`' . $columnName . '`';
            if ($orderSupported && isset($order[$columnName])) {
                $ret .= ' ' . $order[$columnName];
            }

            return $ret;
        }, $columnNames);
```

Reply:
Thanks @MasterOdin